### PR TITLE
[py2py3] Migration at level scr/python/A/B/C - slice 13

### LINKED ISSUE #10137: [py2py3] Migration at level scr/python/A/B/C - slice 13
#### Intro

Our purpose is to continue with the gradual migration. 
This is the 13-th slice in which the migration has been divided into. 
We have split migration units at level `src/python/A/B/C`.

A detailed plan of the migration at this depth is available 
at the dmwm mkdocs [detailed plan page](https://cms-dmwm-test.docs.cern.ch/py2py3/plan_detailed/), 
with summary information on top.

#### Impact of the new feature

We start migrating the 13-th bulk of migration units 
at src/python/A/B/C level.

```plaintext
               name  nfiles  gradual  future_pm
WMCore_WMSpec_Steps      51    False        100
```

#### Is your feature request related to a problem? Please describe.

The end goal is to have WMCore py2 and py3 compatible.

#### Describe the solution you'd like

Run futurize on the first bulk of units 
and then apply the necessary manual changes.

### PR DESCRIPTION
**Pull request by mapellidario**
Fixes #10137 

#### Status
ready

#### Related PRs

* Previous migration PR: #10307 (slice12, issue #10136  )
* Next migration PR: #10313  (slice14, issue #10138 )

#### Description

Run futurize and some manual changes on the first batch of src/python/A/B/C.

Kept back:
- `src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py` has been mostly kepy back, because #9926 and #9934 are the correct places to address how DQMupload uses with urllib

#### Is it backward compatible (if not, which system it affects?)

It should be. (Any possible cause for errors will we reported here)

#### External dependencies / deployment changes

Requires python-future in both py2 and py3 environments.

### CODE REVIEW

**Comment by mapellidario** [comment]
The warning in the section "Pylint py3k check:" comes from `DQMUpload.py`, that has not been fully modernized here, since it is dealt with in for #9926 / #9934 instead (I might go back to that issue tomorrow or next week)

File: test/python/WMCore_t/WMSpec_t/Steps_t/Executors_t/LogArch_t.py [hunk 1]
```diff
@@ -5,6 +5,14 @@
 """
 from __future__ import print_function
 
+try:
+    # https://pylint.pycqa.org/en/latest/technical_reference/features.html
+    # W1626: the `reload` built-in function is missing in python3
+    # we can use imp.reload (deprecated) or importlib.reload
+    from importlib import reload
```
Comment (amaltaro) [inline test/python/WMCore_t/WMSpec_t/Steps_t/Executors_t/LogArch_t.py:12]: Nice catch!

**Review by amaltaro — APPROVED** [review]
Looks pretty good to me, thanks Dario!

File: test/python/WMCore_t/WMSpec_t/Steps_t/Executors_t/LogArch_t.py (hunk 1 above)
Comment (mapellidario) [reply test/python/WMCore_t/WMSpec_t/Steps_t/Executors_t/LogArch_t.py:12]: You have to thank `pylint --py3k`! :)

**Pull request merged**